test/software/amazon/ionschema/IonSchemaImportTest.kt

```diff
@@ -0,0 +1,43 @@
+package software.amazon.ionschema
+
+import org.junit.Assert.*
+import org.junit.Test
+import software.amazon.ion.system.IonSystemBuilder
+
+class IonSchemaImportTest {
+    private val ION = IonSystemBuilder.standard().build()
+    private val iss = IonSchemaSystemBuilder.standard()
+            .addAuthority(AuthorityFilesystem("data/test"))
+            .build()
+
+    @Test
+    fun getImportedType() {
+        assertNotNull(iss.loadSchema("schema/Customer.isl").getType("positive_int"))
+    }
+
+    @Test
+    fun getTypes() {
+        val types = iss
+                .loadSchema("schema/Customer.isl")
+                .getTypes()
+                .asSequence()
+                .associateBy({ it.name() })
+        assertTrue(types.contains("Customer"))
+        assertTrue(types.contains("positive_int"))
+    }
+
+    @Test
+    fun validateType() {
+        val type = iss.loadSchema("schema/Customer.isl").getType("Customer")!!
+        assertTrue(type.isValid(ION.singleValue("{ firstName: \"Phil\", lastName: \"Collins\" }")))
+        assertFalse(type.isValid(ION.singleValue("{ firstName: \"Phil\", middleName: \"Billy\" }")))
+    }
+
+    @Test
+    fun validateAgainstImportedType() {
```

Comment:
same comment as above for this test method

Reply:
Although we are effectively just testing the validation logic, I believe there is value in testing the validation against an imported type  -- this verifies that we do import, we can validate against it. More generally, this is a way to test that the type has been imported successfully.

I do agree that the test above may be less valuable.

Reply:
Agreed, there's value in testing validation against an imported type.  However, the changes you've proposed in `data/test/schema/Customer.isl` do exactly that.  A successful test run against that file requires loading the Customer type, which in turn imports `positive_int` from `CommonTypes.isl`, and then verifying that the age field validates as expected.  It exercises the exact same set of API calls and codepaths that this test does.

Do we need unit tests that exercise API behavior?  Absolutely.  But we should avoid creating unit tests that duplicate those within `data/test`.